### PR TITLE
Responsive Toolbar

### LINKED ISSUE #1393: Re-Org Toolbar for Summary View

### PR DESCRIPTION
**Pull request by jameskerr**
fixes #1393 

1. Makes the existing toolbar responsive (buttons hide when the space get's small)
2. Removes irrelevant buttons when in "summary" view.

I spent some time organizing the commits into logical chunks. Stepping through those might be a good way to review.

3iOuTPsRc6

Bxxq3a2zUm

### CODE REVIEW

File: app/toolbar/hooks/usePackets.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  useDebouncedEffect(() => setEnabled(!!conn && space.pcap_support), 100, [
+    conn,
+    space.pcap_support
+  ])
+
+  function getTitle() {
+    if (!space.pcap_support) return "This space has no pcap support."
+    if (conn) return "Open packets from this connection."
+    else return "No connection record found."
```
Comment (mason-fish) [inline app/toolbar/hooks/usePackets.ts:24]: nit: don't need this `else`

File: app/columns/columns-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  return ReactDOM.createPortal(
+    <CSSTransition
+      in={true}
+      classNames="dim-portal-overlay"
+      timeout={{enter: 200}}
+      onClick={onClose}
+      appear
+    >
+      <div className="portal-overlay">
+        <div className="portal-item" style={style}>
```
Comment (mason-fish) [inline app/columns/columns-modal.tsx:29]: Is there any way we can style this all with just styled-components?

File: app/toolbar/actions.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import useColumns from "./hooks/useColumns"
+import useExport from "./hooks/useExport"
+import usePackets from "./hooks/usePackets"
+import useView from "./hooks/useView"
+import ResponsiveActions from "./responsive-actions"
+
+export default function Actions() {
+  const mainView = useSelector(Layout.getMainView)
+  const showSummary = useSelector(Feature.show("summary"))
+  const view = useView()
```
Comment (mason-fish) [inline app/toolbar/actions.tsx:15]: Nice, separating those hooks out really makes this clean 👌

File: app/toolbar/responsive-actions.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+      (c) => c.getBoundingClientRect().width
+    )
+  }, [parent])
+
+  const onResize = useCallback(
+    ({width}) => {
+      let total = 0
+      let index = 0
+      for (let childWidth of widths.current) {
+        total += childWidth + 8
```
Comment (mason-fish) [inline app/toolbar/responsive-actions.tsx:66]: mind labeling this magic number `8`?

**Review by mason-fish — APPROVED** [review]
Checked this out and it looks awesome!! 🔥 

I left a couple small comments, also I think you have an empty `useMainView.ts` file that you don't need ;)

File: app/columns/columns-modal.tsx (hunk 2 above)
Comment (jameskerr) [reply app/columns/columns-modal.tsx:29]: I don't want this Portal Overlay to live much longer. This is the only place it's used, so I've added a deprecation comment. Next time we need to add features to the column picker, I'd like to delete this "portal" component. And at that time with Styled Components. Just didn't seem worth it to convert it.

File: app/toolbar/responsive-actions.tsx (hunk 4 above)
Comment (jameskerr) [reply app/toolbar/responsive-actions.tsx:66]: Yeah totally, thanks for pointing that out.

**Pull request merged**